stock_owner_restriction/models/stock_move.py

```diff
@@ -1,14 +1,30 @@
 # Copyright 2020 Carlos Dauden - Tecnativa
 # Copyright 2020 Sergio Teruel - Tecnativa
+# Copyright 2023-2024 Quartile
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 from collections import defaultdict
 
-from odoo import models
+from odoo import api, fields, models
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    # Add compute method to the standard field
+    restrict_partner_id = fields.Many2one(
+        compute="_compute_restrict_partner_id",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("picking_type_id", "picking_id.owner_id", "move_dest_ids")
+    def _compute_restrict_partner_id(self):
+        for move in self:
+            if move.picking_type_id.owner_restriction == "picking_partner":
```

Comment:
Please check the validity of this suggestion.
```suggestion
            if move.picking_type_id.owner_restriction in ("picking_partner", "partner_or_unassigned"):
```

Reply:
I think this is not valid. Even if we have move lines from an unassigned owner together with move lines from an assigned owner, the move will still be assigned to the partner. Therefore, we should continue allowing False for this restriction type.